web/src/stream_edit.js

```diff
@@ -665,21 +666,35 @@ export function initialize() {
         return true;
     });
 
+    const save_stream_privacy_settings = ($save_button) => {
+        const $subsection_elem = $save_button.closest(".settings-subsection-parent");
+
+        const stream_id = $save_button.closest(".subscription_settings.show").data("stream-id");
+        const sub = sub_store.get(stream_id);
+        const data = settings_org.populate_data_for_request($subsection_elem, false, sub);
+
+        const url = "/json/streams/" + stream_id;
+        settings_org.save_organization_settings(data, $save_button, url);
+    };
+
     $("#streams_overlay_container").on(
         "click",
         ".subsection-header .subsection-changes-save button",
         (e) => {
             e.preventDefault();
             e.stopPropagation();
             const $save_button = $(e.currentTarget);
-            const $subsection_elem = $save_button.closest(".settings-subsection-parent");
-
-            const stream_id = $save_button.closest(".subscription_settings.show").data("stream-id");
-            const sub = sub_store.get(stream_id);
-            const data = settings_org.populate_data_for_request($subsection_elem, false, sub);
 
-            const url = "/json/streams/" + stream_id;
-            settings_org.save_organization_settings(data, $save_button, url);
+            dialog_widget.launch({
```

Comment:
Does this use the dialog widget even when the change is to settings other than stream privacy? 

The section contains several other settings.